timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -1342,6 +1347,32 @@ double py_accumulate_energy(const py::array_t<long long, py::array::c_style> &in
     return static_cast<long long>(res[0]);
 }
 
+template <typename RealType>
+py::array_t<double, py::array::c_style> py_rotate_coords(
+    const py::array_t<double, py::array::c_style> &coords, const py::array_t<double, py::array::c_style> &quaternions) {
+    verify_coords(coords);
+
+    size_t quaternions_ndims = coords.ndim();
+    if (quaternions_ndims != 2) {
+        throw std::runtime_error("quaternions dimensions must be 2");
+    }
+    if (quaternions.shape(quaternions_ndims - 1) != 4) {
+        throw std::runtime_error("quaternions must have a shape that is 4 dimensional");
+    }
+
+    std::vector<RealType> v_quaternions(quaternions.size());
+    for (int i = 0; i < quaternions.size(); i++) {
+        v_quaternions[i] = static_cast<RealType>(quaternions.data()[i]);
+    }
```

Comment:
Could this use `py_array_to_vector`?

Reply:
No, because this takes doubles and converts them to floats if `RealType=float`. Could separate this out into a different function for converting from an input type to an output type though.

Reply:
Renamed to better match `py_array_to_vector` https://github.com/proteneer/timemachine/pull/1174/commits/be05c972548bbaeab7ea50bcd51ec6d01fa8c857

Reply:
+1 for `py_array_to_vector_with_cast`. Could also see generalizing the existing `py_array_to_vector` to add a second type parameter (but would require updating existing usage and in most places the input and output types would be the same).